x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/utils/LoggingDeprecationAccumulationHandler.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.core.ml.utils;
+
+import org.apache.logging.log4j.LogManager;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.common.logging.LoggerMessageFormat;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Very similar to {@link org.elasticsearch.common.xcontent.LoggingDeprecationHandler} main differences are:
+ * 1. Is not a Singleton
+ * 2. Accumulates all deprecation warnings into a list that can be retrieved
+ *    with {@link LoggingDeprecationAccumulationHandler#getDeprecations()}
+ *
+ * NOTE: The accumulation is NOT THREAD SAFE
+ */
+public class LoggingDeprecationAccumulationHandler implements DeprecationHandler {
+    /**
+     * The logger to which to send deprecation messages.
+     */
+    private static final DeprecationLogger deprecationLogger = new DeprecationLogger(LogManager.getLogger(ParseField.class));
```

Comment:
If it's possible to directly forward to `LoggingDeprecationHandler.INSTANCE` I'd rather do that because it would avoid duplicating the knowledge that the logger for `ParseField.class` goes to the deprecation log.